### DISCUSSION TITLE
When to use this instead of nom?

### DISCUSSION THREAD

**Discussion by clarkmcc** [zesterer/chumsky#406] [Q&A] [answered]
I'm just diving into the world of parsers and am wondering where I should spend my time learning, chumsky or [nom](https://github.com/rust-bakery/nom)?

I'm working on a Common Expression Language interpreter and would like to try implementing it in something like this instead.

What are the reasons I'd go with this vs nom?

**Comment by zesterer** [comment, marked as answer]
`nom` is designed more for binary and other machine-readable formats. If you want to parse a file format, `nom` is a good choice. However, it's not designed for human-readable formats and includes very slim pickings when it comes to error messages and error recovery. If you're looking to write a parser for a human-readable language, like a programming language, chumsky might fit the bill better, since it's capable of generating rich and useful error messages for users.

**Reply by dave-doty** [reply]
Is there somewhere where it is explained how to generate useful error messages? This page seems abandoned: https://github.com/zesterer/chumsky/blob/main/guide/error_and_recovery.md But if there are examples somewhere, that would be a big help, I just don't know where to look.

**Reply by zesterer** [reply]
> Is there somewhere where it is explained how to generate useful error messages? This page seems abandoned: https://github.com/zesterer/chumsky/blob/main/guide/error_and_recovery.md But if there are examples somewhere, that would be a big help, I just don't know where to look.

It does need more documentation, certainly. The `nano_rust.rs` and `json.rs` examples do demonstrate non-trivial error generation and recovery though, you might find them useful.
